CK3toEU4/Source/Mappers/RulerPersonalitiesMapper/RulerPersonalitiesMapper.h

```diff
@@ -0,0 +1,31 @@
+#ifndef RULER_PERSONALITY_MAPPER_H
+#define RULER_PERSONALITY_MAPPER_H
+
+#include "RulerPersonalitiesMapping.h"
+#include "Parser.h"
+#include <set>
+
+namespace CK3
+{
+class Character;
+}
+
+namespace mappers
+{
+class RulerPersonalitiesMapper: commonItems::parser
+{
+  public:
+	RulerPersonalitiesMapper();
+	explicit RulerPersonalitiesMapper(std::istream& theStream);
+
+	[[nodiscard]] std::set<std::string> evaluatePersonalities(const std::pair<int, std::shared_ptr<CK3::Character>>& theCharacter) const;
```

Comment:
You only use the latter part of that pair, so just take the character as input. Better yet, you only use the traits from the character, so take just the traits (unless that pushes too much abstraction up to the callers).

Reply:
I know and would agree, but I find that shoving entire pairs is much cleaner in the functions that do actual stuff. I treat characters, titles, provinces and everything else as pairs and in 95% cases I need both elements. 

So when I say "I need personalities of this character" then the syntax .evaluatePersonalities(character) simply looks better than .evaluatePersonalities(character->second->getTraits());

Noone looks in the mappers anyway once they pass tests, while the code upstairs gets reshuffled quite often.